lib/active_merchant/billing/gateways/priority.rb

```diff
@@ -0,0 +1,471 @@
+      end
+
+      def basic_auth(key, secret)
+        Base64.strict_encode64("#{key}:#{secret}")
+      end
+
+      def request_headers(options)
+        {
+          'Content-Type' => 'application/json',
+          'Authorization' => "Basic #{basic_auth(options[:key], options[:secret])}"
+        }
+      end
+
+      def request_verify_headers
+        auth = test? ? self.test_auth : self.live_auth
+        env_verify = test? ? self.test_env_verify : self.live_env_verify
+        {
+          'Authorization' => auth.to_s,
+          'EOS-Virtual-Environment' => env_verify.to_s
+        }
+      end
+
+      def purchase(amount, credit_card, options = {})
+        # byebug
+        params = {}
+        action = 'purchase'
+        authOnly = false
+        isSettleFunds = true
+        add_bank_amount_purchase(params, amount, authOnly)
+        add_credit_card(params, credit_card, action)
+        add_type_merchant_purchase(params, options[:merchant], isSettleFunds, options)
+        commit(action, params, '', '', options)
+      end
+
+      def authorize(amount, credit_card, options = {})
+        #   byebug
+        params = {}
+        action = 'purchase'
+        authOnly = true
+        isSettleFunds = false
+        add_bank_amount_purchase(params, amount, authOnly)
+        add_credit_card(params, credit_card, action)
+        add_type_merchant_purchase(params, options[:merchant], isSettleFunds, options)
+        commit(action, params, '', '', options)
+      end
+
+      def refund(amount, credit_card, authCode, options, key_secret)
+        #   byebug
+        params = {}
+        action = 'refund'
+        add_bank_amount_refund(params, amount, authCode)
+        add_credit_card(params, credit_card, action)
+        add_type_merchant_refund(params, options)
+        commit(action, params, '', '', key_secret)
+      end
+
+      def capture(amount, authorization, authCode, options = {})
+        #   byebug
+        params = {}
+        action = 'capture'
+        add_capture(params, amount, authorization, authCode, options)
+        commit(action, params, '', '', options)
+      end
+
+      def add_capture(params, amount, authorization, authCode, options)
+        # byebug
+        params['amount'] = amount
+        params['authCode'] = authCode
+        params['merchantId'] = options[:merchant]
+        params['paymentToken'] = authorization
+        params['shouldGetCreditCardLevel'] = true
+        params['source'] = 'Spreedly'
+        params['tenderType'] = 'Card'
+      end
+
+      def void(iid, options)
+        # byebug
+        action = 'void'
+        commit(action, '', iid, '', options)
+      end
+
+      def verify(creditcardnumber)
+        # byebug
+        #   test = request_verify_headers
+        action = 'verify'
+        commit(action, '', '', creditcardnumber, '')
+      end
+
+      def getpaymentstatus(batchid, options)
+        action = 'getpaymentstatus'
+        commit(action, batchid, '', '', options)
```

Comment:
You can set this `action` param to 'getpaymentstatus' and remove the cached variable